src/decide.ts

```diff
@@ -227,84 +267,89 @@ function releaseGroupDoesMatch(
 }
 
 async function assessCandidateHelper(
-	{ link, size, name, tracker }: Candidate,
+	candidate: Candidate,
 	searchee: Searchee,
 	hashesToExclude: string[],
 ): Promise<ResultAssessment> {
 	const { matchMode, blockList } = getRuntimeConfig();
-	if (findBlockedStringInReleaseMaybe(searchee, blockList)) {
-		return { decision: Decision.BLOCKED_RELEASE };
-	}
-
-	if (size && !sizeDoesMatch(size, searchee)) {
-		return { decision: Decision.SIZE_MISMATCH };
-	}
-
-	if (!link) return { decision: Decision.NO_DOWNLOAD_LINK };
+	const { name, size, link } = candidate;
 
 	if (!releaseGroupDoesMatch(searchee.name, name, matchMode)) {
 		return { decision: Decision.RELEASE_GROUP_MISMATCH };
 	}
 	if (!releaseVersionDoesMatch(searchee.name, name, matchMode)) {
 		return { decision: Decision.PROPER_REPACK_MISMATCH };
 	}
-	const result = await snatch(link, tracker);
+	if (size) {
+		if (!fuzzySizeDoesMatch(size, searchee)) {
+			return { decision: Decision.FUZZY_SIZE_MISMATCH };
+		}
+	} else {
+		if (!resDoesMatch(searchee.name, name, matchMode)) {
+			return { decision: Decision.RESOLUTION_MISMATCH };
+		}
+	}
+	if (!link) {
+		return { decision: Decision.NO_DOWNLOAD_LINK };
+	}
+
+	if (findBlockedStringInReleaseMaybe(searchee, blockList)) {
+		return { decision: Decision.BLOCKED_RELEASE };
+	}
```

Comment:
Due to how the decisions will be retried in the cache branch, it's better if we can reject it by the other reasons. Decisions like group, repack, and resolution will never be retried since they can't change. Fuzzy won't be unless the value is now higher.

I also think blocklist further down makes more sense to limit its occurrence. The group not matching means it will never be a cross seed so it should be reported first. So blocklist decisions can be interpreted as "passed all checks before snatch".

Reply:
good points 👍